### PR TITLE
Improve v2 performance for token account balance migration

### LINKED ISSUE #7672: TokenAccountBalanceMigration performs poorly in v2
### Problem

This migration struggles to run in v2

### Solution

Improve migration to run in a reasonable amount of time in v2

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by edwin-greene**
**Description**:
Increases the performance of the token account balance migration with v2

- Fix token balance snapshot CTE to use deduped balance info
- Split the query to reduce the number of timestamp partitions to scan

**Related issue(s)**:

Fixes #7672 

**Notes for reviewer**:
Tested on the Citus cluster with a timestamp range returning ~4 million token balance rows. The migration (without inserting, just selecting the rows to be inserted) completed in 10 to 15 seconds. 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/TokenAccountBalanceMigration.java [hunk 1]
```diff
@@ -90,13 +103,29 @@ public String getDescription() {
...
 
     @Override
     protected void doMigrate() {
+        var count = new AtomicInteger();
         var stopwatch = Stopwatch.createStarted();
-        int count = jdbcOperations.update(UPDATE_TOKEN_ACCOUNT_SQL);
+        transactionTemplate.executeWithoutResult(s -> {
+            try {
+                var timestampInfo = jdbcTemplate.queryForObject(
+                        GET_TIMESTAMP_INFO_SQL, Collections.emptyMap(), new DataClassRowMapper<>(TimestampInfo.class));
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/TokenAccountBalanceMigration.java:116]: can you create an abstract base class, move  GET_TIMESTAMP_INFO_SQL and the inner record `TimestampInfo` over so the code becomes shared by both`InitializeEntityBalanceMigration` and this migration?

Comment (edwin-greene) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/TokenAccountBalanceMigration.java]: Sounds good, I have updated it with an abstract class that combines the shared functionality.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by mgoelswirlds — APPROVED** [review]
LGTM

**Pull request merged**